src/platform/posix/tcp_ip_bundle.c

```diff
@@ -141,13 +145,16 @@ PortMessage *TcpIpBundle_receive(TcpIpBundle *self) {
   }
 
   self->read_index += bytes_read;
-  pb_istream_t stream = pb_istream_from_buffer(self->read_buffer, self->read_index);
 
-  if (!pb_decode(&stream, PortMessage_fields, &self->output)) {
-    printf("decoding failed: %s\n", stream.errmsg);
+  int bytes_left = decode_protobuf(&self->output, self->read_buffer, self->read_index);
+
+  if (bytes_left < 0) {
     return NULL;
```

Comment:
I think we want to change the function type to return `lf_ret_t` and take a `PortMessage**` as an argument so we can get some info back, e.g. that the other side disconnected

Reply:
I dont think there is anything to gain from returning `lf_ret_t` because, there are only to possible responses `LF_OK` or `LF_ERR` I don't get more detailed feedback. So I would just stick with this more simplistic interface.